tests/test_study.py

```diff
@@ -679,6 +679,35 @@ def test_append_trial(storage_mode):
         assert study.best_value == 0.8
 
 
+@pytest.mark.parametrize('storage_mode', STORAGE_MODES)
+def test_enqueue_trial(storage_mode):
+    # type: (str) -> None
+
+    with StorageSupplier(storage_mode) as storage:
+        study = optuna.create_study(storage=storage)
+        assert len(study.trials) == 0
+
+        study.enqueue_trial(params={'x': -1, 'y': 1})
+        assert study._pop_waiting_trial_id() is not None
+        assert study._pop_waiting_trial_id() is None
+
+
+@pytest.mark.parametrize('storage_mode', STORAGE_MODES)
+def test_enqueue_trial_properly_sets_param_values(storage_mode):
+    # type: (str) -> None
+
+    with StorageSupplier(storage_mode) as storage:
+        study = optuna.create_study(storage=storage)
+        assert len(study.trials) == 0
+
+        study.enqueue_trial(params={'x': -1, 'y': 1})
+        trial_id = study._pop_waiting_trial_id()
+        assert trial_id is not None
+        trial = optuna.Trial(study, trial_id)
+        assert trial.suggest_int('x', low=-10, high=10) == -1
+        assert trial.suggest_int('y', low=-10, high=10) == 1
```

Comment:
Can we also test 
- calling suggest with parameters (`name`s) that are not enqueued (and just assert that valid parameters are sampled)?
- the other two parameters to `enqueue_trial`?